### ISSUE TITLE
Rework current workspace approach to facilitate site working directory approach

### ISSUE THREAD

**Issue by tannerjfco** [drud/ddev#47]
## What happened (or feature request):
Feature Request

TL;DR: I want to `cd ~/mysites/my-site-project` and run `ddev start` instead of `ddev start [sitename]` and `cd ~/.drud/plugin/sitename/src`

Our tool currently works by defining its own workspace, whose default is `~/.drud`. In this workspace, sites are organized by plugin, and each site has its folder structure generated when the site is added.

## What you expected to happen:
We desire to have a workflow that allows the user to be in control of where there site files are on the host machine. This aligns with the desire for the user to be in control of their source checkouts, where we were previously controlling all resources to provision the site.

## Anything else do we need to know:
This will heavily impact the current workspace implementation and the `workon` command. I suspect the `workon` command would be deprecated with this change.

## Related source links or issues (like source JIRA issue):
#27
